ext/reflection/php_reflection.c

```diff
@@ -6392,18 +6404,35 @@ ZEND_METHOD(ReflectionAttribute, getName)
 }
 /* }}} */
 
-static zend_always_inline int import_attribute_value(zval *ret, zval *val, zend_class_entry *scope) /* {{{ */
+/* {{{ proto public int ReflectionAttribute::getTarget()
+ *	   Returns the target of the attribute */
+ZEND_METHOD(ReflectionAttribute, getTarget)
 {
-	ZVAL_COPY_OR_DUP(ret, val);
+	reflection_object *intern;
+	attribute_reference *attr;
 
-	if (Z_TYPE_P(val) == IS_CONSTANT_AST) {
-		if (SUCCESS != zval_update_constant_ex(ret, scope)) {
-			zval_ptr_dtor(ret);
-			return FAILURE;
-		}
+	if (zend_parse_parameters_none() == FAILURE) {
+		RETURN_THROWS();
 	}
+	GET_REFLECTION_OBJECT_PTR(attr);
 
-	return SUCCESS;
+	RETURN_LONG(attr->target);
+}
+/* }}} */
+
+/* {{{ proto public bool ReflectionAttribute::isRepeated()
+ *	   Returns true if the attribute is repeated */
+ZEND_METHOD(ReflectionAttribute, isRepeated)
+{
+	reflection_object *intern;
+	attribute_reference *attr;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		RETURN_THROWS();
+	}
+	GET_REFLECTION_OBJECT_PTR(attr);
+
+	RETURN_BOOL(zend_is_attribute_repeated(attr->attributes, attr->data));
```

Comment:
These APIs were not mentioned in the RFC :(

Reply:
(The APIs themselves seem like reasonable additions though.)

Reply:
Can we add these APIs or should I remove them from the PR? Both methods can be replaced with userland code instead. The target could be known if the user has access to the reflector. Looking for repeated attributes would simply need to iterate the attribute list, the C implementation does the same thing.

Reply:
I think these additions are fine.